### DISCUSSION TITLE
--mix-str with filter?

### DISCUSSION THREAD

**Discussion by kr1tx** [dashingsoft/pyarmor#1380] [Q&A]
- What does --mix-str with filter mean?

**Comment by jondy** [comment]
If special string constants need not be mixed but all the others need, filter these special strings.

Refer to
https://pyarmor.readthedocs.io/en/stable/tutorial/advanced.html#filter-mix-string

**Comment by kr1tx** [comment]
I have used these options `pyarmor gen -e 2023-09-20 --enable-bcc --enable-rft --enable-jit --mix-str --assert-import --obf-code 2 --private --pack dist\Script.exe Script.py` but can't see any sign of mixed strings in pyarmor.trace.log

**Comment by kr1tx** [comment]
if I enable RFT, this means i don't need --mix-str?

**Comment by jondy** [comment]
RFT doesn't touch string constants. `--mix-str` is required if need to encrypt string constants.

**Comment by kr1tx** [comment]
I have used these options `pyarmor gen -e 2023-09-20 --enable-bcc --enable-rft --enable-jit --mix-str --assert-import --obf-code 2 --private --pack dist\Script.exe Script.py` but can't see any sign of mixed strings in pyarmor.trace.log`

**Comment by jondy** [comment]
Try to use `--mix-str` only, and append other options one by one. 

By default `--mix-str` only encrypts the string length > 8.
 
Besides `--enable-bcc` may convert Python code to c, the strings in these functions are ignored by `--mix-str`

**Comment by kr1tx** [comment]
I have already referred to https://pyarmor.readthedocs.io/en/stable/tutorial/advanced.html#filter-mix-string
this is what my config file contain: 
```
[mix.str]
includes = /.{4,500}/
threshold = 3
excludes = __main__

```

**Use --mix-str only and append other options one by one**
how?

**Comment by jondy** [comment]
First `pyarmor gen --mix-str foo.py`, check the trace log
Then `pyarmor gen --mix-str --enable-rft foo.py`, check the trace log
...

**Comment by kr1tx** [comment]
works fine now thanks
